bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusBaseHandler.java

```diff
@@ -66,41 +72,45 @@ public void initialize() {
             return;
         }
         int pollingInterval = getPollingInterval(appliance);
-        scheduler.scheduleWithFixedDelay(this::updateChannels, 0, pollingInterval, TimeUnit.SECONDS);
-
-        updateStatus(ThingStatus.UNKNOWN);
+        if (poller != null) {
+            // Cancel any previous polling
+            poller.cancel(true);
+        }
+        poller = scheduler.scheduleWithFixedDelay(this::updateChannels, thingCounter, pollingInterval,
+                TimeUnit.SECONDS);
+        logger.debug("Scheduled polling every {}s for appliance {}", pollingInterval, thing.getUID());
     }
 
     @Override
-    public void channelLinked(ChannelUID channelUID) {
-        super.channelLinked(channelUID);
-
-        OndusService ondusService = getOndusService();
-        if (ondusService == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
-                    "No initialized OndusService available from bridge.");
-            return;
+    public void dispose() {
+        logger.debug("Disposing scheduled updater for thing {}", thing.getUID());
+        if (poller != null) {
+            poller.cancel(true);
         }
+        super.dispose();
+    }
 
-        @Nullable
-        T appliance = getAppliance(ondusService);
-        if (appliance == null) {
-            return;
-        }
-        updateChannel(channelUID, appliance, getLastDataPoint(appliance));
+    @Override
+    public void initialize() {
+        config = getConfigAs(GroheOndusApplianceConfiguration.class);
+        schedulePolling();
     }
 
     public void updateChannels() {
+        logger.debug("Updating channels for appliance {}", thing.getUID());
         OndusService ondusService = getOndusService();
         if (ondusService == null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
                     "No initialized OndusService available from bridge.");
```

Comment:
Consider moving text to src/main/resources/OH-INF/i18n/groheondus.properties to allow for localization.